frontend/www/js/omegaup/components/course/Card.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div class="col mb-3">
+    <div class="card">
+      <div class="row no-gutters">
+        <div class="col-sm-2 col-lg-3" :class="`${type}-course-card`"></div>
+        <div class="col-sm-10 col-lg-9">
+          <div class="card-body">
+            <h5 class="card-title">{{ course.name }}</h5>
+            <p v-if="type === 'public'" class="card-text">
+              {{ course.description }}
+            </p>
+            <omegaup-markdown
+              v-if="type !== 'finished'"
+              class="card-text"
+              :markdown="
+                ui.formatString(T.courseCardImpartedBy, {
+                  school_name: course.school_name,
+                })
+              "
+            ></omegaup-markdown>
+            <div
+              v-if="type === 'student'"
+              class="row no-gutters justify-content-start align-items-center mb-4"
+            >
+              <div class="col-4">{{ T.wordsProgress }}:</div>
+              <div class="col-8 mt-1">
+                <div class="progress">
+                  <div
+                    class="progress-bar progress-bar-striped bg-warning text-dark"
+                    role="progressbar"
+                    :style="`width: ${course.progress}%`"
+                    :aria-valuenow="course.progress"
+                    aria-valuemin="0"
+                    aria-valuemax="100"
+                  >
+                    {{ `${course.progress}%` }}
+                  </div>
+                </div>
+              </div>
+            </div>
+            <div
+              v-if="type === 'finished'"
+              class="text-center mb-4 course-star"
+            >
+              ⭐
+            </div>
+            <div class="text-center">
+              <a
+                class="btn btn-primary text-white"
+                role="button"
+                :href="`/course/${course.alias}/`"
+                >{{ buttonMessage }}</a
+              >
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import T from '../../lang';
+import * as ui from '../../ui';
+
+import omegaup_Markdown from '../Markdown.vue';
+
+@Component({
+  components: {
+    'omegaup-markdown': omegaup_Markdown,
+  },
+})
+export default class CourseCard extends Vue {
+  @Prop() course!: types.FilteredCourse;
+  @Prop() type!: string;
```

Comment:
perdón, debí haber notado esto antes, pero después de revisar https://github.com/omegaup/omegaup/pull/5075/files#diff-d43d535d75287222e2a6b0e3a5e18e7afcfde89e1c330dae04ecb98ac8eb3078R5, será posible usar un enum aquí también?

Reply:
No estoy seguro de si lo hice como querías, me avisas.

Reply:
```suggestion
  @Prop() type!: CourseType;
```

faltaría esto y cambiar las literales de L91, L93, L95 a `CourseType.Finished` y sus amigos.